### PR TITLE
[WIP] All-in-one Doc Updates

### PR DESCRIPTION
**Pull request by jfryman**
This PR contains updates to our documentation to include details about the AIO, as well as how to programatically deploy it.

This is not done, but uploading for comment. It needs to be converted from MD -> RST, but I didn't want formatting to stop getting the actual content out.

Early comments appreciated.

### CODE REVIEW

**Comment by dzimine** [comment]
the direction (and what you have so far) is LGTM.

One thought it to flash out what IS all-in-one installer (takes some reflection) my take is: it's a "production ready reference deployment scaled down to one box for easy evaluation, simple production, and learning how stackstorm components are deployed / wired together)". @jfryman think about it and do your take. if we don't have time for this, np.

**Comment by jfryman** [comment]
@dzimine will certainly add more, but in the dot release. going for MVP here. Great feedback though. Made myself a sticky note. :notebook:

File: docs/source/install/vagrant.rst [hunk 1]
```diff
@@ -1,9 +1,83 @@
...
+We are constantly striving to ensure that we have compatability with as many platforms as we can. However, if you find a basebox that doesn't work, please let us know and we'll be glad to take a look. We also love Pull Requests from the community, and might have some goodies if you help us out!
+
+Sample Vagrantfile
+------------------
+
+Below is an an example of a Vagrantfile capable of loading StackStorm. This minimal Vagrantfile will load up a machine and provision a clean StackStorm installation.This configuration also includes a `public_network` setting, which is necessary to allow your host environment to access the StackStorm guest machine. If you choose not to use this configuration, make sure that you have an interface configured that you can access via the Host Machine.
+
+::
+
+    Vagrant.configure(2) do |config|
```
Comment (dzimine) [inline docs/source/install/vagrant.rst]: @manasdk had some HOST vooodo in his vagrant to make it work? or work better? See https://github.com/manasdk/st2-test-ground/blob/master/Vagrantfile#L4-L5

if it's nessessary or helpful let's add

**Comment by dzimine** [comment]
awesome!

make sure it passes travis and +1

File: docs/source/install/vagrant.rst (hunk 1 above)
Comment (jfryman) [inline docs/source/install/vagrant.rst]: Nope. It's not necessary. If they want bells and whistles, they can use `st2workroom`, which is exactly what this is designed for. This is reference.

**Comment by dzimine** [comment]
@jfryman hope you won't mind some minor structural changes; and you do please override. To me it looks good enough, when travis passes I'll merge and you can hack on top. 

Scrutinizer seem to give false negatives.

**Comment by jfryman** [comment]
@dzimine thanks for fixing. Was having some local issues building.

**Pull request merged**